design-documents/graph-ql/coverage/catalog.graphqls

```diff
@@ -11,7 +11,10 @@ type Query {
     ): Products
     @resolver(class: "Magento\\CatalogGraphQl\\Model\\Resolver\\Products") @doc(description: "The products query searches for products that match the criteria specified in the search and filter attributes.") @cache(cacheIdentity: "Magento\\CatalogGraphQl\\Model\\Resolver\\Product\\Identity")
     category (
-     id: Int @doc(description: "Id of the category.")
+    # Implementation Note: The resolver should accept either `id` or `categoryInterfaceUID`,
+    # but _must_ return an error when both fields are provided in a request
+     id: Int @deprecated(reason: "Use the `categoryInterfaceUID` argument instead")
+     categoryInterfaceUID: ID @doc(description: "Unique identifier from objects implementing `CategoryInterface`")
```

Comment:
Why not just `uid`? (I think with the description it's clear what it is)

Reply:
Let me try and provide some quick context around the naming.

https://github.com/magento/architecture/blob/master/design-documents/graph-ql/id-improvement-plan.md

> **Terminology**
> **Primary Identifier**: An ID owned by the current Object Type (ex: Product.id)
> **Foreign Identifier**: An ID referencing another Object Type (ex: PaymentMethodInput.code)

We've tried, for object fields, to make it easy/predictable to determine what an `ID` field is referencing, by including the concrete type in the name of the field. The general rule of thumb has been `typename_uid` to reference an `ID` value from another object type.

------

Unfortunately, as you've pointed out, there's 2 problems:

1. When you're doing a simple lookup of a known type like `CategoryTree` with an argument, it's redundant/verbose to specify `category(categoryInterfaceUID: "<value>")`
2. That linked document didn't account for naming conventions when referencing an `ID` field defined in an interface, rather than an object type (which you pointed out in another comment about `Breadcrumb.category_interface_uid`)

For this case (a query on the `Query` root, doing a lookup by ID) I agree it makes sense to just switch to `uid`, since it's obvious to a consumer of the API 👍

Reply:
Pushed update changing arg name to `uid`